paimon-core/src/main/java/org/apache/paimon/io/KeyValueFileReaderFactory.java

```diff
@@ -102,11 +103,23 @@ public RecordReader<KeyValue> createRecordReader(DataFileMeta file) throws IOExc
 
     public RecordReader<KeyValue> createRecordReader(
             long schemaId, String fileName, long fileSize, int level) throws IOException {
+        return createRecordReader(schemaId, fileName, fileSize, level, null);
+    }
+
+    public RecordReader<KeyValue> createRecordReader(
```

Comment:
I think you don't need to modify `paimon-core`, you can do the filter outer.

Reply:
Good point! I updated the PR so we don't apply the filter in the `KeyValueFileReaderFactory`.  

However, I think we still need to modify the `LocalTableQuery` in the `paimon-core` unless I am missing something.